social distancing analyser 3.0.py

```diff
@@ -0,0 +1,303 @@
+    else:
+        a_w = p2[0]
+        a_h = p2[1]
+
+    T = 0
+    try:
+        T=(p2[2][1]-p1[2][1])/(p2[2][0]-p1[2][0])
+    except ZeroDivisionError:
+        T = 1.633123935319537e+16
+    S = T2S(T)
+    C = T2C(T)
+    d_hor = C*c_d
+    d_ver = S*c_d
+    vc_calib_hor = a_w*1.3
+    vc_calib_ver = a_h*0.4*angle_factor
+    c_calib_hor = a_w *1.7
+    c_calib_ver = a_h*0.2*angle_factor
+    # print(p1[2], p2[2],(vc_calib_hor,d_hor),(vc_calib_ver,d_ver))
+    if (0<d_hor<vc_calib_hor and 0<d_ver<vc_calib_ver):
+        return 1
+    elif 0<d_hor<c_calib_hor and 0<d_ver<c_calib_ver:
+        return 2
+    else:
+        return 0
+
+
+labelsPath = "./coco.names"
+LABELS = open(labelsPath).read().strip().split("\n")
+
+np.random.seed(42)
+
+weightsPath = "./yolov3.weights"
+configPath = "./yolov3.cfg"
+
+###### use this for faster processing (caution: slighly lower accuracy) ###########
+
+#weightsPath = "./yolov3-tiny.weights"  
+#configPath = "./yolov3-tiny.cfg"
+
+
+net = cv2.dnn.readNetFromDarknet(configPath, weightsPath)
+ln = net.getLayerNames()
+ln = [ln[i[0] - 1] for i in net.getUnconnectedOutLayers()]
+FR=0
+vs = cv2.VideoCapture(vid_path)
+
+
+#Counting total Frames present in given video
+frameCnt = 0
+print("Enter Gap Factor in range (1-4): \n1: Higest Accuracy, Slowest Speed ")
+print("2: High Accuracy, Slow Speed \n3: High Accuracy, Better Speed \n4: Medium Accuracy, Max Speed ")
+gapFactor = int(input("Enter Gap Factor (1 to 4) : "))
+#gapFactor = 3    #number of frames gap to add to incease speed.
+
+while True:
+    (grabbed, frame) = vs.read()
+
+    if not grabbed:
+        #end = time.time()
+        break
+    
+    frameCnt = frameCnt + 1
+
+    #if writer is None:
+    #    fourcc = cv2.VideoWriter_fourcc(*"MP4V")
+    #    writer = cv2.VideoWriter("op_"+vname, fourcc, 30, (frame.shape[1], frame.shape[0]), True)
+    
+    #writer.write(frame)
+
+print("Total Frame Count is ", frameCnt)
+
+vs = cv2.VideoCapture(vid_path)
+
+counter = 0
+# This counter is for counting each 
+
+# vs = cv2.VideoCapture(0)  ## USe this if you want to use webcam feed
+writer = None
+(W, H) = (None, None)
+
+fl = 0
+q = 0
+while True:
+
+    (grabbed, frame) = vs.read()
+    if not grabbed:
+        break
+
+    counter = counter + 1
+
+
+    #frame = cv2.resize(frame, (480,320))
+    frame = cv2.resize(frame, (720,480))
+    #frame = cv2.resize(frame, (720,480))
+    if W is None or H is None:
+        (H, W) = frame.shape[:2]
+        FW=W
+        if(W<1075):
+            FW = 1075
+        FR = np.zeros((H+210,FW,3), np.uint8)
+
+        col = (255,255,255)
+        FH = H + 210
+    FR[:] = col
+
+    #Processing Only Certain gap frames.
+
+    if counter!=1 and counter%gapFactor!=0:
+        frame = cv2.resize(frame, (720,480))
+        #cv2.imshow('Social distancing analyser', frame)
+        #cv2.waitKey(1)
+        #print("Non-Processed Frame ",counter)
+        #writer.write(frame)
+        layerOutputs = oldlayer
```

Comment:
the ide shows undefined name "oldlayer" here.

Reply:
Thank you for your review, **just resolved this**, fixed some minor issues, (like file not found) and as the processing of each frame depends upon the processor so made some changes in the estimated time calculation. 😄